### PR TITLE
Add CI workflow to publish releases

### PR DESCRIPTION
**Pull request by MatteoPologruto**
On every push of a tag named with a version format:

- Build the project for all supported platforms.
- Use [gon](https://github.com/mitchellh/gon) to sign and notarize the macOS builds.
- Create a [GitHub release](https://docs.github.com/repositories/releasing-projects-on-github/about-releases).
  - Builds and checksums are attached as release assets
  - A changelog generated by [`arduino/create-changelog`](https://github.com/arduino/create-changelog) from the commit history is added to the release description
  - If the tag has [a pre-release version suffix](https://semver.org/), the GitHub release will be marked as a pre-release.
- Upload the builds to Arduino's downloads server.

### Additional information
The workflow has been modified to create parallel builds for each OS and calculate the relative checksums once, at the end of the process.
Test run: https://github.com/MatteoPologruto/arduinoOTA/actions/runs/3487993513
Test release: https://github.com/MatteoPologruto/arduinoOTA/releases/tag/99.99.99

### CODE REVIEW

**Review by umbynos — CHANGES_REQUESTED** [review]
I think we can remove also `deploy.sh`.

something is wrong with the version:
```
./arduinoOTA -version
1.3.0
```

**Comment by MatteoPologruto** [comment]
> I think we can remove also `deploy.sh`.
> 
> something is wrong with the version:
> 
> ```
> ./arduinoOTA -version
> 1.3.0
> ```

Working on it.

File: deploy.sh [hunk 1]
```diff
@@ -1,46 +0,0 @@
-#!/bin/bash -xe
```
Comment (umbynos) [inline deploy.sh:1]: If we are removing this, and this is the only shell script, maybe we should remove also https://github.com/arduino/arduinoOTA/blob/ce63040f98ce22778810220179c0d9f234c6a440/.github/workflows/check-shell-task.yml

**Review by umbynos — CHANGES_REQUESTED** [review]

**Comment by umbynos** [comment]
Version printing LGTM

**Review by umbynos — APPROVED** [review]

**Comment by umbynos** [comment]
The check markdown links is expected to fail, while the check markdown nope. @per1234 do you have an idea on why is failing?

**Comment by MatteoPologruto** [comment]
> The check markdown links is expected to fail, while the check markdown nope. @per1234 do you have an idea on why is failing?

@umbynos I think it's because there are some problems with the schema download and the `file-url` is different from the [file url](https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/check-taskfiles.yml#L55) used in the assets repo. I could fix it in a separate PR.

**Comment by umbynos** [comment]
Oh right https://github.com/arduino/tooling-project-assets/commit/888fcbfefee4bc0af483f50610067c336ffd286e :man_facepalming:

**Review by facchinm — APPROVED** [review]

**Pull request merged**